frontend/app/services/subscription-service.server.ts

```diff
@@ -9,16 +9,22 @@ import { getLogger } from '~/utils/logging.server';
 const log = getLogger('subscription-service.server');
 
 const subscriptionInfoSchema = z.object({
-  id: z.string(),
-  userId: z.string(),
+  id: z.string().optional(),
+  userId: z.string().optional(),
   msLanguageCode: z.string(),
   alertTypeCode: z.string(),
-  email: z.string(),
-  emailVerifed: z.boolean(),
+  email: z.string().optional(),
+  emailVerifed: z.boolean().optional(),
 });
 
 type SubscriptionInfo = z.infer<typeof subscriptionInfoSchema>;
 
+const userInfoSchema = z.object({
+  email: z.string(),
+});
+
+type UserInfo = z.infer<typeof userInfoSchema>;
+
 const userSchema = z.object({
```

Comment:
I feel we should come up with another name to distinguish the intended use of `userInfoSchema` vs `userSchema`.